packages/gatsby-source-contentful/src/plugin-options.js

```diff
@@ -32,6 +32,7 @@ const optionsSchema = Joi.object().keys({
   downloadLocal: Joi.boolean(),
   localeFilter: Joi.func(),
   forceFullSync: Joi.boolean(),
+  proxy: Joi.object(),
```

Comment:
This should be more strict
Support for proxy are the following fields:
```
 proxy: {
    host: '127.0.0.1',
    port: 9000,
    auth: {
      username: 'mikeymike',
      password: 'rapunz3l'
    }
  },
```
```suggestion
  proxy: Joi.object().keys({
    host: Joi.string().required(),
    port:  Joi.number().required(),
    auth:  Joi.object().keys({
      username: Joi.string(),
      password: Joi.string(),
    }),
  }),
```